api/services/FootprintService.js

```diff
@@ -0,0 +1,274 @@
+    }
+
+    let child             = manager.getEntities()[relation.targetEntity];
+    let childPk           = child.mapping.getPrimaryKey();
+    let repository        = manager.getRepository(parentModelName);
+    let queryBuilder      = repository.getQueryBuilder('p');
+    let childQueryBuilder = queryBuilder.where({'p': parentId}).populate(childAttributeName);
+    let childAlias        = childQueryBuilder.getAlias();
+    criteria              = typeof criteria === 'string' || typeof criteria === 'number'
+      ? {[childPk]: criteria} : criteria;
+
+    repository.applyOptions(childQueryBuilder, options);
+
+    childQueryBuilder.select(childAlias).where(criteria);
+
+    return queryBuilder.getQuery().getResult();
+  }
+
+  /**
+   *
+   * Update association(s) based on given primary key or criteria
+   *
+   * @param {string}        parentModelName
+   * @param {string|number} parentId
+   * @param {string}        childAttributeName
+   * @param {{}}            criteria
+   * @param {{}}            values
+   * @param {{}}            [options]
+   *
+   * @returns {Promise.<T>}
+   */
+  updateAssociation(parentModelName, parentId, childAttributeName, criteria, values, options = {}) {
+    if (!values) {
+      return this.app.log.error('Missing values to update.');
+    }
+
+    return this.findAssociation(parentModelName, parentId, childAttributeName, criteria, options).then(result => {
+      result            = result[0];
+      let manager       = this.app.orm.getManager();
+      let populator     = this.app.orm.getPopulator(manager);
+      let parent        = manager.getEntities()[parentModelName];
+      let relation      = parent.mapping.getRelation(childAttributeName);
+      let childEntity   = manager.getEntity(relation.targetEntity);
+      options.recursive = options.recursive || 1;
+
+      let updatedChild = Array.isArray(result[childAttributeName])
+        ? result[childAttributeName].map(child => populator.assign(childEntity, values, child, options.recursive - 1))
+        : populator.assign(childEntity, values, result[childAttributeName], options.recursive - 1);
+
+      populator.assign(parent.entity, {[childAttributeName]: updatedChild}, result, options.recursive);
+
+      return manager.flush().then(() => result);
+    });
+  }
+
+  /**
+   * Destroy association(s) based on given primary key or criteria
+   *
+   * @param {string}        parentModelName
+   * @param {string|number} parentId
+   * @param {string}        childAttributeName
+   * @param {{}}            criteria
+   * @param {{}}            [options]
+   *
+   * @returns {Promise.<T>}
+   */
+  destroyAssociation(parentModelName, parentId, childAttributeName, criteria, options) {
+    return this.findAssociation(parentModelName, parentId, childAttributeName, criteria, options).then(result => {
+      if (!result) {
+        return this.app.log.info('No record found with the specified criteria.');
```

Comment:
You can log errors if you want but you must return a Promise or in case of error it will log and crash. Look here for example when the model is not found https://github.com/trailsjs/trailpack-sequelize/blob/master/api/services/FootprintService.js#L44